test/smoketest/src/smoketest/core.clj

```diff
@@ -0,0 +1,48 @@
+(ns smoketest.core
+  (:require [clojure.tools.nrepl :as nrepl]
+            [clojure.tools.nrepl.server :as server])
+  (:gen-class))
+
+;; see cider-nrepl issue #447
+(defn nrepl-handler []
+  (require 'cider.nrepl)
+  (ns-resolve 'cider.nrepl 'cider-nrepl-handler))
+
+(defn nrepl-version
+  "Launch nrepl server and send :cider-version message. This confirms
+  the repl launched with the middleware."
+  []
+  (with-open [serv (server/start-server :handler (nrepl-handler))]
+    (with-open [transport (nrepl/connect :port (:port serv))]
+      (-> (nrepl/client transport 1000)
+          (nrepl/message {:id 2 :op :cider-version})
+          doall
+          first
+          :cider-version
+          :version-string))))
+
+(defn project-version
+  "Get project version assuming first line of project.clj contains
+  the form (def VERSION \"0.17.0-SNAPSHOT\")."
```

Comment:
Seems like some copy/paste mistake to me. :-)

Reply:
All I meant to say is I'm parsing this line
https://github.com/clojure-emacs/cider-nrepl/blob/master/project.clj#L1

Reply:
Ah, for some reason I thought you were doing to parse the `project.clj` for the test project. Btw, might be better to use some middleware that won't require such an extra step - e.g. you can just check the eldoc of something or the classpath or whatever.